hash-db/src/lib.rs

```diff
@@ -20,24 +20,32 @@
 use std::fmt::Debug;
 #[cfg(feature = "std")]
 use std::hash;
+#[cfg(not(feature = "std"))]
+use core::hash;
+
 #[cfg(feature = "std")]
 pub trait MaybeDebug: Debug {}
 #[cfg(feature = "std")]
 impl<T: Debug> MaybeDebug for T {}
-
-#[cfg(not(feature = "std"))]
-use core::hash;
 #[cfg(not(feature = "std"))]
 pub trait MaybeDebug {}
 #[cfg(not(feature = "std"))]
 impl<T> MaybeDebug for T {}
 
 
-/// Empty prefix constant.
-pub static EMPTY_PREFIX: Prefix<'static> = &[];
+/// An empty prefix constant. Mainly use for comparison
+/// and for root nodes.
+pub static EMPTY_PREFIX: Prefix<'static> = (&[], (0,0));
 
-/// Prefix byte information for avoding rc.
-pub type Prefix<'a> = &'a[u8];
+/// The Prefix of a trie node. Prefix is nibble path up to
+/// the node in the trie.
+/// For a value node it is the value key without the Partial
+/// bytes (encoded in the node).
+/// Being the leftmost portion of a byte, its internal representation
+/// is byte slice and and a padded byte.
+/// The padded byte is represented as a pair with number of nibble first,
+/// and left aligned padded value second.
```

Comment:
This is a bit confusing. Why do we need to keep an additional nibble index here?

Reply:
I realize the additional nibble index does not need to be use in the prefix, but it seems like it can be good to have all info, especially since it is use for a trait (for comparing prefix it is needed but that is not really needed).

It also allows building nibble from prefix (strictly speaking it should not be necessary as prefix is either slice and padded byte from key or from a nibblevec).